saas/mixins.py

```diff
@@ -631,6 +643,14 @@ def get_context_data(self, **kwargs):
             context.update({'ends_at': self.ends_at})
         return context
 
+    @property
+    def period_func(self):
+        # Returns the period function from the request with a default
+        # set to month_periods
+        period = self.request.GET.get('period')
+        return self.PERIOD_FUNC_MAP.get(period, month_periods)
```

Comment:
This might need a default value in case `period` is not passed as a query argument.

Reply:
Shouldn't 
"return self.PERIOD_FUNC_MAP.get(period, month_periods)" 
be sufficient? If it's None, it returns month_periods